src/utils/read-pacts.ts

```diff
@@ -2,19 +2,40 @@ import * as fs from 'fs';
 import * as path from 'path';
 
 export function readPacts(dir: string): Pact[] {
-    const fileNames = fs.readdirSync(dir);
+    const filePaths = fs
+        .readdirSync(dir)
+        .map((fileName) => path.join(dir, fileName));
 
-    if (fileNames.length === 0) {
+    if (filePaths.length === 0) {
         throw new Error(`Pact directory ${dir} is empty.`);
     }
 
-    return fileNames.map((fileName) => {
-        const filePath = path.join(dir, fileName);
-        return JSON.parse(fs.readFileSync(filePath, 'utf8'));
+    const pactFiles = filePaths.map((filePath) => {
```

Comment:
Could we merge this map() with code from line #5 ? The result will be the same.

```ts
const filePaths = fs
        .readdirSync(dir)
        .map((fileName) => path.join(dir, fileName))
        .map((filePath) => {
               return {
            pact: JSON.parse(fs.readFileSync(filePath, 'utf8')) as Pact,
            path: filePath,
        };
    });
```
??

Reply:
Merged and removed one of the map functions - joining path is done during mapping.